### PR TITLE
Fix all clippy warnings replay

### PR DESCRIPTION
**Pull request by pinkforest**
Replaying https://github.com/dalek-cryptography/curve25519-dalek/pull/421/files

Will re-base with Co-authored by for @nox

### CODE REVIEW

**Review by rozbb — COMMENTED** [review]
Nice, thanks! Just checked with [`cargo public-api`](https://github.com/Enselic/cargo-public-api) and it looks like there's no breakage introduced by this. I'll impl Hash, do an fmt, and add clippy+fmt to the CI

**Comment by rozbb** [comment]
I don't know why CI isn't running here, but anyways it's now failing. Turns out there's more stuff that clippy's upset about https://github.com/rozbb/curve25519-dalek/actions/runs/3557280501/jobs/5975177321

File: src/montgomery.rs [hunk 1]
```diff
@@ -66,7 +66,8 @@ use zeroize::Zeroize;
 
 /// Holds the \\(u\\)-coordinate of a point on the Montgomery form of
 /// Curve25519 or its twist.
-#[derive(Copy, Clone, Debug, Hash)]
+#[allow(clippy::derive_hash_xor_eq)]
+#[derive(Copy, Clone, Debug, Default, Hash)]
```
[the remark this answers is not in the document]
Comment (rozbb) [reply src/montgomery.rs]: Done

**Comment by pinkforest** [comment]
> I don't know why CI isn't running here,

Because the branch has `/` in it - it has to be '**' for to match branches with `/` - ik confusing :)

**Comment by pinkforest** [comment]
Happy clippy is the best clippy (also CI works) :tada:

**Pull request merged**